packages-ts/gauntlet-terra-contracts/src/lib/contracts.ts

```diff
@@ -28,109 +30,191 @@ export type TerraABI = {
   [TERRA_OPERATIONS.QUERY]: JSONSchemaType<any>
 }
 
-export type Contract = {
-  id: CONTRACT_LIST
+export abstract class Contract {
+  // Contract metadata, initialized in constructor
+  readonly id: CONTRACT_LIST
+  readonly defaultVersion: string
+  readonly dirName: string
+  readonly downloadUrl: string
+
+  // Only load bytecode & schema later if needed
+  version: string
   abi: TerraABI
   bytecode: string
-}
 
-export type Contracts = Record<CONTRACT_LIST, Contract>
+  constructor(id, dirName, defaultVersion) {
+    this.id = id
+    this.defaultVersion = defaultVersion
+    this.dirName = dirName
+  }
+
+  addInstance(envVar: string, name: string = this.id) {
```

Comment:
I added the 3rd parameter mostly for `access_controller`, to distinguish `billing_access` from `requester_access`. But since `cw20_base` and `cw3_flex_multisig` both happen to have simple recognizable aliases, I used it for them as well.

Given our current set of contracts, that leaves `cw4_group` as the only one actually using it.  But it still seems like a convenient default to have, for any other contracts we add that don't need special aliases.